packages/web/src/features/translation/TranslationSidebar.tsx

```diff
@@ -62,32 +62,32 @@ export const TranslationSidebar = ({
     id: language,
   });
 
-  // Only update the content shown in the text editor when the word ID changes.
-  const [originalNoteContent, setOriginalNoteContent] = useState('');
   const [noteContent, setNoteContent] = useState('');
   const wordId = useRef('');
   useEffect(() => {
-    if (word.id !== wordId.current) {
+    if (translatorNotesQuery.isSuccess && word.id !== wordId.current) {
       wordId.current = word.id;
-      setOriginalNoteContent(translatorNote?.content ?? '');
-      setNoteContent(translatorNote?.content ?? '');
+      setNoteContent(translatorNotesQuery.data.data[word.id]?.content ?? '');
     }
-  }, [word.id, translatorNote]);
+  }, [word.id, translatorNotesQuery]);
 
-  const saveNote = useCallback(async () => {
-    await apiClient.words.updateTranslatorNote({
-      wordId: word.id,
-      language,
-      note: noteContent,
-    });
-    translatorNotesQuery.refetch();
-  }, [language, translatorNotesQuery, word.id, noteContent]);
-
-  // eslint-disable-next-line react-hooks/exhaustive-deps
-  const debouncedSaveNote = useCallback(
-    throttle(saveNote, 15000, { leading: false, trailing: true }),
-    [language, translatorNotesQuery, word.id, noteContent]
+  const saveNote = useMemo(
+    () =>
+      throttle(
+        async (noteContent: string) => {
```

Comment:
This idea here is that we have a single function `saveNote` that takes the note content and executes every 15 seconds with the latest note content.